modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt

```diff
@@ -75,35 +81,89 @@ class GolandRunConfigurationExtension : GoRunConfigurationExtension() {
     ) {
         val service = configuration.getProject().service<MirrordProjectService>()
 
-        if (commandLineType == GoRunningState.CommandLineType.RUN &&
-            service.enabled &&
-            SystemInfo.isMac &&
-            state.isDebug
-        ) {
-            val delvePath = getCustomDelvePath()
-            // convert the delve file to an executable
-            val delveExecutable = Paths.get(delvePath).toFile()
-            if (delveExecutable.exists()) {
-                if (!delveExecutable.canExecute()) {
-                    delveExecutable.setExecutable(true)
-                }
-                // parameters returns a copy, and we can't modify it so need to reset it.
-                val patchedParameters = executor.parameters
-                for (i in 0 until patchedParameters.size) {
-                    // no way to reset the whole commandline, so we just remove each entry.
-                    executor.withoutParameter(0)
-                    val value = patchedParameters[i]
-                    if (value.toPresentableString().endsWith("/dlv", true)) {
-                        patchedParameters[i] = PathParameter(delveExecutable.toString())
-                    }
-                }
-                executor.withParameters(*patchedParameters.toTypedArray())
+        if (commandLineType != GoRunningState.CommandLineType.RUN || !service.enabled || !SystemInfo.isMac || !state.isDebug) {
+            super.patchExecutor(configuration, runnerSettings, executor, runnerId, state, commandLineType)
+            return
+        }
+
+        val ourDelvePath = getCustomDelvePath()
+        val delveExecutable = Paths.get(ourDelvePath).toFile()
+        if (!delveExecutable.exists()) {
+            MirrordLogger.logger.error("Failed to find delve bundled with mirrord plugin ($ourDelvePath)")
+            super.patchExecutor(configuration, runnerSettings, executor, runnerId, state, commandLineType)
+            return
+        }
+
+        if (!delveExecutable.canExecute()) {
+            delveExecutable.setExecutable(true)
+        }
+
+        val ourDelveVersion = try {
+            getDelveVersion(ourDelvePath)
+        } catch (e: Throwable) {
+            MirrordLogger.logger.error("Failed to get version delve bundled with mirrord plugin ($ourDelvePath)", e)
+            super.patchExecutor(configuration, runnerSettings, executor, runnerId, state, commandLineType)
+            return
+        }
+
+        // parameters returns a copy, and we can't modify it so need to reset it.
+        val patchedParameters = executor.parameters
+        for (i in 0 until patchedParameters.size) {
+            // no way to reset the whole commandline, so we just remove each entry.
+            executor.withoutParameter(0)
+
+            val foundDelvePath = patchedParameters[i].toPresentableString()
+
+            if (!foundDelvePath.endsWith("/dlv", true)) {
```

Comment:
Is is feesable that there will be a relative path here? like a singular string of `dlv`?

Reply:
Not sure, I left this check as it was. Have been working fine so far :sweat_smile: